conda_env/installers/pip.py

```diff
@@ -56,6 +64,8 @@ def _pip_install_via_requirements(prefix, specs, args, *_, **kwargs):
         # So we delete the temporary file in a finally block.
         if requirements is not None and op.isfile(requirements.name):
             os.remove(requirements.name)
+        if script_caller is not None:
+            os.remove(script_caller)
```

Comment:
As @kalefranz mentioned in #8229 this should be `rm_rf` not `os.remove` unless I am missing something.  If you want to change the call in line ~1065 of `link.py` that would be great.